api/office_structure/models.py

```diff
@@ -38,3 +38,27 @@ def add_branch(self, nodes):
         db_session.add_all(nodes)
 
         db_session.commit()
+
+    @property
+    def nested_children(self):
+        """
+        Function to return a dictionary of the nested children
```

Comment:
Hello @RachaelNantale , good work.  I think this function returns a list, not a dictionary. Kindly confirm this. If it's true, please update the docstring.

Reply:
To add on, Is it possible to write tests for these functions?

Reply:
Thanks, @nakatuddesuzan  for the feedback. so for your first question. This returns a dictionary

Reply:
Okay, I hadn't noticed the use `retrieve_nested_data` function here.

Reply:
hey @nakatuddesuzan, I have realised that since this function is being used when querying all locations, it has already been covered.